numpy/core/tests/test_arrayprint.py

```diff
@@ -524,6 +524,22 @@ def test_refcount(self):
         gc.enable()
         assert_(r1 == r2)
 
+    def test_with_sign(self):
+        a = np.array([-2, 0, 3])
+        assert_equal(
+            np.array2string(a, sign='+'),
+            '[-2 +0 +3]'
+        )
+        assert_equal(
+            np.array2string(a, sign='-'),
+            '[-2  0  3]'
+        )
+        assert_equal(
+            np.array2string(a, sign=' '),
+            '[-2  0  3]'
+        )
```

Comment:
To tell the intended difference between `sign='-'` and `sign=' '`, use an all-positive array. See the behavior for floating point inputs as a reference.

```python
>>> np.array2string(np.array([2., 0., 3.]), sign='-')
'[2. 0. 3.]'

>>> np.array2string(np.array([2., 0., 3.]), sign=' ')
'[ 2.  0.  3.]'
```

Reply:
Awesome, I will fix all of the suggested changes.  Grateful for the feedback!

Reply:
Fixed the tests in order to demonstrate the desired behavior.